accounts/abi/reflect.go

```diff
@@ -90,7 +82,11 @@ func set(dst, src reflect.Value) error {
 	case srcType.AssignableTo(dstType) && dst.CanSet():
 		dst.Set(src)
 	case dstType.Kind() == reflect.Slice && srcType.Kind() == reflect.Slice && dst.CanSet():
-		setSlice(dst, src)
+		return setSlice(dst, src)
+	case dstType.Kind() == reflect.Array && dst.CanSet():
```

Comment:
I suggest you check for `dst.CanSet()` inside `setArray()` so that you return a more specific error